### PR TITLE
feat: confirm method for the emitter

### LINKED ISSUE #130: Require a confirm prompts
If the library can provide something like [Click: Confirm Prompts](https://click.palletsprojects.com/en/8.1.x/prompts/#confirmation-prompts) will be really useful.

### PR DESCRIPTION
**Pull request by lengau**
This adds a confirm method to the emitter, allowing the program to ask the user a yes/no question.

Fixes #130 

- [ ] Have you followed the guidelines for contributing?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `tox`?

### CODE REVIEW

File: craft_cli/messages.py [hunk 1]
```diff
@@ -781,6 +781,29 @@ def set_secrets(self, secrets: list[str]) -> None:
         """Set the list of strings that should be masked out in all output."""
         self._printer.set_secrets(secrets)
 
+    @_active_guard()
+    def confirm(self, prompt: str, *, default: bool = False) -> bool:
```
Comment (tigarmo) [inline craft_cli/messages.py:785]: is there any change here from the charmcraft version? Other than the `self.pause()`

Comment (tigarmo) [inline craft_cli/messages.py:785]: please add a changelog entry
then we can expedite thisss

File: tests/unit/test_messages_emitter.py [hunk 2]
```diff
@@ -1167,3 +1177,70 @@ def test_reporterror_command_error_no_stderr(get_initiated_emitter, stderr):
...
+
+    assert emit.confirm("prompt", default=True) is True
+    assert emit.confirm("prompt", default=False) is False
+
+    assert mock_input.mock_calls == []
+
+
+@pytest.mark.parametrize(
+    ("user_input", "expected"),
+    [
```
Comment (mr-cal) [inline tests/unit/test_messages_emitter.py:1214]: How about some test cases with leading whitespace to exercise `strip()`?

**Review by mr-cal — APPROVED** [review]
Looks good!

File: tests/unit/test_messages_emitter.py (hunk 2 above)
Comment (lengau) [reply tests/unit/test_messages_emitter.py:1214]: Done!

File: craft_cli/messages.py (hunk 1 above)
Comment (lengau) [reply craft_cli/messages.py:785]: Dun!

**Review by tigarmo — APPROVED** [review]

**Review by medubelko — APPROVED** [review]

**Pull request merged**